packages/devtools/lib/src/timeline/cpu_flame_chart.dart

```diff
@@ -0,0 +1,169 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:async';
+import 'dart:math' as math;
+import '../ui/drag_scroll.dart';
+import '../ui/fake_flutter/dart_ui/dart_ui.dart';
+import 'cpu_profile_protocol.dart';
+import 'flame_chart.dart';
+
+final StreamController<CpuFlameChartItem> _selectedCpuFlameChartItemController =
+    StreamController<CpuFlameChartItem>.broadcast();
+
+Stream<CpuFlameChartItem> get onSelectedCpuFlameChartItem =>
+    _selectedCpuFlameChartItemController.stream;
+
+final DragScroll _dragScroll = DragScroll();
+
+// TODO(kenzie): get feedback on this palette and add dark theme colors.
+const colorPalette = [
+  Color(0xFFFF5722),
+  Color(0xFFFF6D00),
+  Color(0xFFFFAB00),
+  Color(0xFFFFD600),
+  Color(0xFFFFEA00),
+];
+
+class CpuFlameChart extends FlameChart<CpuProfileData> {
+  CpuFlameChart()
+      : super(
+          onSelectedFlameChartItem: onSelectedCpuFlameChartItem,
+          dragScroll: _dragScroll,
+          classes: 'cpu-flame-chart ui-details-section',
+          flameChartInset: 4,
+        );
+
+  static const samplePadding = 1;
+
+  int _colorOffset = 0;
+
+  // TODO(kenzie): do something more intelligent for color selection (maybe base
```

Comment:
update TODO: perhaps colors should be based on categories, e.g. Widget, Render, Layer, and user code.

Reply:
done.